amp-cli/amp
# Updates for PHP 8.4

General notes:

* This adds some fixes for warnings that live directly in `amp.git`.
* Additionally, it updates a bunch of dependencies - which resolves warnings in them.
* If there were some problem, then I'd be most suspicious of `symfony/config` and `symfony/dependency-injection` because they have a relatively long list of changes for 4.x=>5.x. But...
* I held `symfony/process` back on ~4.4 instead upgrading to 5.x. It does have some contract-changes that would require code-updates. However, so far, `symfony/process:~4.4` seems to be working fine with php84.

QA:

* Executed `run-tests.sh` locally, and all combinations passed -- including the new `php84m80` test-run.
    * (Of course, upstream `phpunit9` still has its own warnings on php84. These involve `Prophecy` classes - but that doesn't affect the ability to run or test `amp`.)
* Did some manual testing, eg
    * Play around with `config:set`, `config:get`, `config` and ensure that the configuration-files are actually updated as expected. (*This is because I was curious about the size of the upgrades for `symfony/config` and `symfony/dependency-injection`.*)
    * Put a new `amp.phar` into my copy of buildkit. Ran commands which involve `amp`... such as`civibuild create`, `civibuild reinstall`, `civibuild `restore`, and `setup.sh`.
        * In php73, the overall process still works  (as you'd hope/expect).
        * In php84, the overall process has problems.  I don't think these are in `amp` per se. Rather, a bunch of other projects -- D7 core, `drush`, `civicrm-core`, etc -- all have their own struggles with php84.

So, that seems like progress... `amp` shouldn't be the limiting factor in php84 usage...
